src/Orleans.Runtime/Streams/QueueBalancer/LeaseBasedQueueBalancer.cs

```diff
@@ -51,9 +51,10 @@ public RoundRobinSelector(ReadOnlyCollection<T> resources)
         public List<T> NextSelection(int newSelectionCount, List<T> existingSelection)
         {
             var selection = new List<T>(newSelectionCount);
-            while (selection.Count < newSelectionCount)
+            int tries = 0;
+            while (selection.Count < newSelectionCount && tries++ < this.resources.Count)
             {
-                this.lastSelection = (++this.lastSelection) % (this.resources.Count - 1);
+                this.lastSelection = (++this.lastSelection) % (this.resources.Count);
```

Comment:
Please sanity check this change.  I believe the previous logic would never allow for the last resource to be allocated.
It also allowed infinite loop if more resource were requested than were available.

Reply:
You are probably right. But can we add some tests testing this. I forgot why I added that minus 1, but I remember I do spent sometime on this. 
Add tests : 
- NextSelectionWillGoThroughEveryResourceIfExistingSelectionEmpty
- NextSelectionWontGoInfinitely
- NextSelectionWontReSelectExisitngSelections